lalrpop/src/tok/test.rs

```diff
@@ -509,3 +509,10 @@ fn char_literals() {
         (r#"                          ~~"#, Lifetime("'c")),
     ]);
 }
+
+#[test]
+fn code_blocks() {
+    let input = r#" {do_something(); println!('{'); println!('\''); }, { other_block(); } "#;
+    let mut tokenizer = Tokenizer::new(&input, 0);
+    assert_eq!(tokenizer.code(0, "([{", "}])"), Ok(50));
```

Comment:
👍 to a test, but can't we use the `test!` infrastructure for this? something like

```rust
#[test]
fn code_blocks() {
    test(r#" {println!('\'');}, {other_block();} "#, vec![
        (r#" ~~~~~~~~~~~~~~~~~ ", ...)]);
}
```

That would avoid hardcoding indices and things, and (of course) make things more uniform.

Reply:
The existing `test` just seemed very token-oriented to me, whereas code is only ever really expected to return an integer once.

That said I agree it'd be nice to do something similar (but separate) with the same tilde underlining to specify expected matches. If we just merge this fix as-is I'll open and work on an issue to expand the `code` tests to something more exhaustive and we can worry about that there.

Sound good?

Reply:
Hmm, I'm not sure I understand. How is this not a token-oriented test? `code` is just an internal helper function, but it's "effects" bubble up and ought to be visible in terms of the tokenization, right? Is it that you prefer to test the `code` function individually, rather than the output of the tokenizer? Put another way, I like the way the tests are written now, and this doesn't (yet) seem like an improvement to me, so I'm wondering what I'm missing. =)

Reply:
Ah, I think I see what you want instead - check out the test as it is now in the latest commit.